### PR TITLE
Show native validation message on blur

### PR DESCRIPTION
**Pull request by Ararat2**

### CODE REVIEW

File: src/Fields/Checkbox/Checkbox.js [hunk 1]
```diff
@@ -39,10 +41,15 @@ const Checkbox = ({ id, initial, value: checkboxValue, ...props }) => {
...
-            id
-          })
+        onBlur={(e) => {
+          if (!validationMessageIsShown) {
+            e.target.reportValidity()
+          }
+
+          setValidationMessageIsShown(validationMessageIsShown => !validationMessageIsShown)
+
+          handleBlur({ id })
```
Comment (Tiko-Nazaryan) [inline src/Fields/Checkbox/Checkbox.js:51]: First of all lets move this to handler function

File: src/Fields/Input/Input.js [hunk 2]
```diff
@@ -41,6 +47,7 @@ const Input = ({ id, type, initial, ...props }) => {
             field: { id, initial, type, ...props }
           })
         }
+        required
```
Comment (Tiko-Nazaryan) [inline src/Fields/Input/Input.js:50]: This parameter should depend from error which we are have in our state. Also we need a pattern either for just showing the validation mssg.

File: src/Fields/Checkbox/Checkbox.js [hunk 3]
```diff
@@ -1,11 +1,13 @@
-import React, { useEffect, useContext } from 'react'
```
Comment (Tiko-Nazaryan) [inline src/Fields/Checkbox/Checkbox.js:1]: I cant see where is the error msgs setting up part

File: src/Store/Reducer.js [hunk 4]
```diff
@@ -83,27 +83,25 @@ export const reduser = (state, action) => {
           else delete errors[name]
           break
         case 'checkbox':
```
Comment (TikoNazaryan) [inline src/Store/Reducer.js:85]: try to just remove this case at all and if it's not work explain me please why

File: src/Store/Reducer.js [hunk 5]
```diff
@@ -170,32 +167,31 @@ export const reduser = (state, action) => {
           break
 
         case 'checkbox':
```
Comment (TikoNazaryan) [inline src/Store/Reducer.js:169]: Same here. Lets just remove this case.

File: src/Store/Reducer.js (hunk 4 above)
Comment (Ararat2) [reply src/Store/Reducer.js:85]: Checkbox field errors are not being added to the errors object. They are handled specific way

File: src/Fields/Checkbox/Checkbox.js [hunk 6]
```diff
@@ -3,15 +3,17 @@ import context from './../../Store/Context'
...
   const { handleChange, handleBlur, declareField } = actions
   const { values, errors } = state
-  const possibleError = errors[props.name]
+  const possibleError = errors[id]
+  const { HTMLValidate = false } = validation
+
+  console.log("STATE", state)
 
   useEffect(() => {
     const actualInitial = {
```
[the remark this answers is not in the document]
Comment (Ararat2) [reply src/Fields/Checkbox/Checkbox.js:17]: FIxed

File: src/Store/Reducer.js (hunk 4 above)
Comment (TikoNazaryan) [reply src/Store/Reducer.js:85]: I see. Can we handle that case in validation function ?. Please try it somehow )

**Pull request merged**